src/main/java/org/jenkinsci/plugins/workflow/support/pickles/serialization/RiverWriter.java

```diff
@@ -107,12 +105,9 @@ public RiverWriter(File f, FlowExecutionOwner _owner) throws IOException {
     public RiverWriter(File f, FlowExecutionOwner _owner, final Collection<? extends PickleFactory> pickleFactories) throws IOException {
         file = f;
         owner = _owner;
-        dout = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(file)));
-        dout.writeLong(HEADER);
-        dout.writeShort(VERSION);
-        ephemeralsBackptr = dout.size();
-        LOGGER.fine(() -> "Starting to save " + file + "; pickle offset will be written @" + ephemeralsBackptr);
-        dout.writeInt(0);     // we'll back-fill this address with the pointer to the ephemerals stream
+        channel = FileChannel.open(file.toPath(), StandardOpenOption.WRITE);
+        channel.write(HEADER_BUFFER.duplicate());
+        LOGGER.fine(() -> "Starting to save " + file + "; pickle offset will be written @" + EPHEMERALS_BACKPTR);
```

Comment:
These log messages from #123 may no longer be helpful, and I never managed to use them to pinpoint a problem in the current state; feel free to delete any that do not seem to be contributing anything.

Reply:
I think they are mostly fine, but I removed the reference to `EPHEMERALS_BACKPTR` here in 9f5f6c4 since it is a constant.